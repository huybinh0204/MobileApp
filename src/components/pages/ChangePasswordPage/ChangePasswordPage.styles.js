import styled from '@emotion/native';
import { BaseText, IconSvg } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  padding: theme.spacing.m,
}));

export const Illustration = styled(IconSvg)(({ theme }) => ({
  marginTop: theme.spacing.m,
}));

export const ParagraphContainerStyled = styled.View({
  flex: 1,
  justifyContent: 'center',
});

export const TextStyled = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  textAlign: 'center',
}));

export const SecondaryTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  paddingVertical: theme.spacing.s,
  textAlign: 'center',
}));
