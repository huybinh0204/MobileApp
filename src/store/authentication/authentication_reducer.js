import { createActions, createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';

const { Types, Creators } = createActions(
  {
    // handled by the authentication_sagas
    logout: ['isUniversalLoginEnabled'],
    signUp: ['email', 'password'],
    signIn: { email: null, password: null, isEmailVerificationEnabled: null },
    resetPassword: ['email'],
    fetchCurrentUserData: [],
    // handled by the authentication_reducer
    setAuthenticationError: ['authenticationError'],
    setCredentials: ['credentials'],
    setCurrentUserData: ['currentUserData'],
    setCurrentUserDataError: ['currentUserDataError'],
    setIsLoading: ['isLoading'],
    setIsLoadingCurrentUserData: ['isLoadingCurrentUserData'],
    setIsLoadingResetPassword: ['isLoadingResetPassword'],
    setIsSessionExpired: ['isSessionExpired'],
    setIsSignIn: ['isSignIn'],
    setResetPasswordError: ['resetPasswordError'],
    setResetPasswordSuccess: ['resetPasswordSuccess'],
    resetAuthentication: ['overrides'],
  },
  {
    prefix: '@@BE-AUTHENTICATION/',
  }
);

export const AuthenticationActions = Creators;
export const AuthenticationTypes = Types;

// INITIAL STATE
export const INITIAL_STATE = Immutable({
  authenticationError: null,
  credentials: null,
  currentUserData: null,
  currentUserDataError: null,
  isLoading: false,
  isLoadingCurrentUserData: false,
  isLoadingResetPassword: false,
  isSessionExpired: false,
  isSignIn: false,
  resetPasswordError: false,
  resetPasswordSuccess: false,
});

// SELECTORS
export const AuthenticationSelectors = {
  getAuthentication: (state) => state.auth.credentials?.accessToken,
  getAuthenticationError: (state) => state.auth.authenticationError,
  getCurrentUserData: (state) => state.auth.currentUserData,
  getCurrentUserDataError: (state) => state.auth.currentUserDataError,
  getCredentials: (state) => state.auth.credentials,
  getIsLoading: (state) => state.auth.isLoading,
  getIsLoadingCurrentUserData: (state) => state.auth.isLoadingCurrentUserData,
  getIsLoadingResetPassword: (state) => state.auth.isLoadingResetPassword,
  getIsSessionExpired: (state) => state.auth.isSessionExpired,
  getIsSignIn: (state) => state.auth.isSignIn,
  getResetPasswordError: (state) => state.auth.resetPasswordError,
  getResetPasswordSuccess: (state) => state.auth.resetPasswordSuccess,
};

// ACTION HANDLERS
const resetAuthentication = (_state, { overrides = {} }) => {
  return INITIAL_STATE.merge({ ...overrides });
};

const setAuthenticationError = (state, { authenticationError }) => {
  return state.merge({ authenticationError });
};

const setCredentials = (state, { credentials }) => {
  return state.merge({ credentials });
};

const setCurrentUserDataError = (state, { currentUserDataError }) => {
  return state.merge({ currentUserDataError });
};

const setCurrentUserData = (state, { currentUserData }) => {
  return state.merge({ currentUserData });
};

const setIsLoading = (state, { isLoading }) => {
  return state.merge({ isLoading });
};

const setIsLoadingCurrentUserData = (state, { isLoadingCurrentUserData }) => {
  return state.merge({ isLoadingCurrentUserData });
};

const setIsLoadingResetPassword = (state, { isLoadingResetPassword }) => {
  return state.merge({ isLoadingResetPassword });
};

const setIsSessionExpired = (state, { isSessionExpired }) => {
  return state.merge({ isSessionExpired });
};

const setIsSignIn = (state, { isSignIn }) => {
  return state.merge({ isSignIn });
};

const setResetPasswordError = (state, { resetPasswordError }) => {
  return state.merge({ resetPasswordError });
};

const setResetPasswordSuccess = (state, { resetPasswordSuccess }) => {
  return state.merge({ resetPasswordSuccess });
};

// REDUCER
export const AuthenticationReducer = createReducer(INITIAL_STATE, {
  [Types.SET_AUTHENTICATION_ERROR]: setAuthenticationError,
  [Types.SET_CREDENTIALS]: setCredentials,
  [Types.SET_CURRENT_USER_DATA]: setCurrentUserData,
  [Types.SET_CURRENT_USER_DATA_ERROR]: setCurrentUserDataError,
  [Types.SET_IS_LOADING]: setIsLoading,
  [Types.SET_IS_LOADING_CURRENT_USER_DATA]: setIsLoadingCurrentUserData,
  [Types.SET_IS_LOADING_RESET_PASSWORD]: setIsLoadingResetPassword,
  [Types.SET_IS_SESSION_EXPIRED]: setIsSessionExpired,
  [Types.SET_IS_SIGN_IN]: setIsSignIn,
  [Types.SET_RESET_PASSWORD_ERROR]: setResetPasswordError,
  [Types.SET_RESET_PASSWORD_SUCCESS]: setResetPasswordSuccess,
  [Types.RESET_AUTHENTICATION]: resetAuthentication,
});
