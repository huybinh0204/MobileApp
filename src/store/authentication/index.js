export * from './authentication_reducer';
export * from './authentication_sagas';
