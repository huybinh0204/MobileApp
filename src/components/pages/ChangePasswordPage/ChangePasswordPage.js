import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { MainButton, Toast } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS, TOAST_TYPES } from '_constants';
import strings from '_localization';
import { AuthenticationActions, AuthenticationSelectors } from '_store/authentication';
import { CustomerSelectors } from '_store/customer';
import { normalize } from '_utilities/screen';
import {
  Container,
  Illustration,
  ParagraphContainerStyled,
  SecondaryTitle,
  TextStyled,
} from './ChangePasswordPage.styles';

const ChangePasswordPage = () => {
  const dispatch = useDispatch();
  const error = useSelector(AuthenticationSelectors.getResetPasswordError);
  const emailSent = useSelector(AuthenticationSelectors.getResetPasswordSuccess);
  const isLoading = useSelector(AuthenticationSelectors.getIsLoadingResetPassword);
  const customer = useSelector(CustomerSelectors.getCustomer);

  const toastHeader = error
    ? strings.forgotPassword.errorTitle
    : strings.forgotPassword.successTitle;

  const toastContent = error
    ? strings.forgotPassword.errorMessage
    : strings.forgotPassword.successMessage;

  const type =
    (error && TOAST_TYPES.ERROR) || (emailSent && TOAST_TYPES.SUCCESS) || TOAST_TYPES.ERROR;

  const handleSubmit = () => {
    dispatch(AuthenticationActions.resetPassword(customer?.email));
  };

  const handleCloseToast = () => {
    dispatch(AuthenticationActions.setResetPasswordError(false));
    dispatch(AuthenticationActions.setResetPasswordSuccess(false));
  };

  return (
    <SecondaryScreenLayout title={strings.settings.change_password} testID="ChangePasswordPage">
      <Container>
        <Illustration icon={ICONS.checkEmail} width={normalize(300)} height={normalize(300)} />
        <ParagraphContainerStyled>
          <SecondaryTitle numberOfLines={1}>{strings.changePassword.title}</SecondaryTitle>
          <TextStyled numberOfLines={2}>{strings.changePassword.subtitle}</TextStyled>
        </ParagraphContainerStyled>
        <MainButton
          accessibilityLabel="Change Password Button"
          testID="changePasswordButton"
          isLoading={isLoading}
          onPress={handleSubmit}
        >
          {strings.changePassword.button}
        </MainButton>
      </Container>
      <Toast
        type={type}
        onClose={handleCloseToast}
        testID="ChangePasswordToast"
        header={toastHeader}
        content={toastContent}
        show={error || emailSent}
      />
    </SecondaryScreenLayout>
  );
};

export default ChangePasswordPage;
