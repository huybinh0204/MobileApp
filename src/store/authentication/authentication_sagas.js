import { Dynatrace } from '@dynatrace/react-native-plugin';
import jwtDecode from 'jwt-decode';
import { call, put, select } from 'redux-saga/effects';
import { AUTH0_ERRORS, EVENTS, EVENT_TYPES, NAVIGATION } from '_constants';
import strings from '_localization';
import AppsFlyerService from '_services/AppsFlyerService';
import BrazeService from '_services/BrazeService';
import { navigate } from '_services/NavigationService';
import { AccountActions } from '_store/account';
import { CustomerActions } from '_store/customer';
import { FundDirectDepositStoreActions } from '_store/pageStore/fundDirectDepositStore';
import { RegisterActions } from '_store/register';
import { TransferActions } from '_store/transfer';
import { ISODateTimeFromNow } from '_utilities/date';
import { shouldReportError } from '_utilities/dynatrace';
import { AuthenticationActions, AuthenticationSelectors } from './authentication_reducer';

export function* signIn(api, { email, password, isEmailVerificationEnabled }) {
  let auth0IdToken = null;

  try {
    yield put(AuthenticationActions.setIsLoading(true));

    const isEmbeddedFlow = email && password;
    const { accessToken, expiresIn, idToken } = isEmbeddedFlow
      ? yield call(api.signIn, email, password) // API call
      : yield call(api.authorize); // Universal Login

    auth0IdToken = idToken;
    const sessionTTL = expiresIn - 30; // add 30 sec margin to prevent the token from expiring during the request
    const expiration = yield call(ISODateTimeFromNow, { seconds: sessionTTL });
    const credentials = yield call(jwtDecode, accessToken);
    const userId = credentials.sub;

    yield call(api.setAuthorization, accessToken, userId);
    yield put(AuthenticationActions.setCredentials({ ...credentials, accessToken, expiration }));
    const { data } = yield call(api.getIdentity, userId);
    const {
      externalId: accountExternalId,
      userExternalId: customerExternalId,
      mustContinueOnboarding,
    } = data;

    if (mustContinueOnboarding) {
      const userData = auth0IdToken ? yield call(jwtDecode, auth0IdToken) : null;
      yield put(CustomerActions.setExternalId(customerExternalId));

      const targetScreen = isEmailVerificationEnabled
        ? NAVIGATION.auth.signUpMailVerification
        : NAVIGATION.auth.signUpPhone;
      yield call(navigate, targetScreen, { email: email ?? userData?.email });

      yield put(RegisterActions.trackEvent(EVENTS.ONBOARDING_RESUMED, EVENT_TYPES.TRACK));
    } else {
      yield call(BrazeService.changeUser, customerExternalId);
      yield call(AppsFlyerService.identify, customerExternalId);
      yield call(Dynatrace.identifyUser, customerExternalId);
      yield put(AccountActions.setExternalId(accountExternalId));
      yield put(CustomerActions.setExternalId(customerExternalId));
      yield put(AuthenticationActions.setIsSignIn(true));
      yield put(RegisterActions.trackEvent(EVENTS.SIGN_IN_SUCCESS, EVENT_TYPES.TRACK));
    }
  } catch (e) {
    if (e?.error === AUTH0_ERRORS.USER_CANCELLED_LOGIN) {
      return;
    }

    const { description, title } = strings.signIn.defaultError;
    let error = e?.errorMessage ?? { title, description: e?.message ?? description };
    if (e?.message === AUTH0_ERRORS.SIGN_IN_OUTSIDE_US) {
      error = strings.signIn.outsideUSError;
    }

    yield put(AuthenticationActions.setAuthenticationError(error));
    if (shouldReportError(e?.status)) {
      yield call(Dynatrace.reportError, `Sign In Error: ${JSON.stringify(e)}`, 0);
    }
  } finally {
    yield put(AuthenticationActions.setIsLoading(false));
  }
}

export function* signUp(api, { email, password }) {
  try {
    yield put(AuthenticationActions.setIsLoading(true));
    yield call(api.signUp, email, password);
    const { accessToken, expiresIn } = yield call(api.signIn, email, password);
    const sessionTTL = expiresIn - 30; // add 30 sec margin to prevent the token from expiring during the request
    const expiration = yield call(ISODateTimeFromNow, { seconds: sessionTTL });
    const credentials = yield call(jwtDecode, accessToken);
    const userId = credentials.sub;
    yield call(api.setAuthorization, accessToken, userId);
    const { data } = yield call(api.setIdentity, userId, email);
    const { userExternalId: customerExternalId } = data;
    yield put(CustomerActions.setExternalId(customerExternalId));
    yield put(AuthenticationActions.setCredentials({ ...credentials, accessToken, expiration }));
    yield put(RegisterActions.trackEvent(EVENTS.AUTH0_SIGN_UP_SUCCESS, EVENT_TYPES.TRACK));
    yield call(AppsFlyerService.track, EVENTS.AF_COMPLETE_AUTH0_REGISTRATION);
  } catch (e) {
    const { description, title } = strings.signUp.authError;
    let error = { title, description: e?.message ?? description };

    if (e?.code === AUTH0_ERRORS.INVALID_SIGN_UP) {
      error = strings.signUp.signUpExistAccountError;
    }

    if (e?.message === AUTH0_ERRORS.NO_SIGN_UP_OUTSIDE_US) {
      error = strings.signUp.signUpBlockSignUpError.outsideUS;
    }
    yield put(AuthenticationActions.setAuthenticationError(error));
    if (shouldReportError(e?.status)) {
      yield call(Dynatrace.reportError, `Sign Up Auth0 Error: ${JSON.stringify(e)}`, 0);
    }
  } finally {
    yield put(AuthenticationActions.setIsLoading(false));
  }
}

export function* fetchCurrentUserData(api) {
  try {
    yield put(AuthenticationActions.setIsLoadingCurrentUserData(true));
    const credentials = yield select(AuthenticationSelectors.getCredentials);
    const accessToken = credentials?.accessToken ?? null;
    const currentUserData = yield call(api.fetchCurrentUserData, accessToken);
    yield put(AuthenticationActions.setCurrentUserData(currentUserData));
  } catch (e) {
    yield put(AuthenticationActions.setCurrentUserDataError(true));
    if (shouldReportError(e?.status)) {
      yield call(Dynatrace.reportError, `Fetch Current User Data Error: ${JSON.stringify(e)}`, 0);
    }
  } finally {
    yield put(AuthenticationActions.setIsLoadingCurrentUserData(false));
  }
}

export function* logout(api, { isUniversalLoginEnabled }) {
  try {
    if (isUniversalLoginEnabled) {
      yield call(api.clearSession);
    }
    yield call(api.clearAuthorization);
    yield put(AuthenticationActions.resetAuthentication());
    yield put(AccountActions.resetAccount());
    yield put(TransferActions.resetTransfer());
    yield put(FundDirectDepositStoreActions.resetRenderData());
    yield call(AppsFlyerService.reset);
  } catch (e) {
    if (e?.error !== AUTH0_ERRORS.USER_CANCELLED_LOGIN) {
      yield call(Dynatrace.reportError, `Sign Out Auth0 Error: ${JSON.stringify(e)}`, 0);
    }
  }
}

export function* resetPassword(api, { email }) {
  try {
    yield put(AuthenticationActions.setIsLoadingResetPassword(true));
    yield call(api.resetPassword, email);
    yield put(AuthenticationActions.setResetPasswordSuccess(true));
    yield put(RegisterActions.trackEvent(EVENTS.RESET_PASSWORD_EMAIL_SENT, EVENT_TYPES.TRACK));
  } catch (e) {
    yield put(AuthenticationActions.setResetPasswordError(true));
    if (shouldReportError(e?.status)) {
      yield call(Dynatrace.reportError, `Reset Password Error: ${JSON.stringify(e)}`, 0);
    }
  } finally {
    yield put(AuthenticationActions.setIsLoadingResetPassword(false));
  }
}
