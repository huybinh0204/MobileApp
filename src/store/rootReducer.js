import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
