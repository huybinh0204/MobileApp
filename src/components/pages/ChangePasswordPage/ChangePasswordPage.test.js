import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { createStore } from 'redux';
import strings from '_localization/index';
import { AuthenticationActions } from '_store/authentication/authentication_reducer';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import ChangePasswordPage from './ChangePasswordPage';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('ChangePasswordPage', () => {
  let mockStore;
  let mockedDispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(mockedDispatch);
    mockStore = createStore(rootReducer, {
      customer: {
        data: {
          email: 'user@lendup.com',
        },
      },
      auth: {
        resetPasswordSuccess: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={ChangePasswordPage} store={mockStore} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should trigger the change password flow when change password button is pressed', async () => {
    const { getByText } = render(
      <ComponentWithProviders component={ChangePasswordPage} store={mockStore} />
    );

    const changePasswordButton = getByText(strings.changePassword.button);

    await waitFor(() => {
      fireEvent.press(changePasswordButton);
    });

    expect(mockedDispatch).toHaveBeenCalledWith(
      AuthenticationActions.resetPassword('user@lendup.com')
    );
  });
});
