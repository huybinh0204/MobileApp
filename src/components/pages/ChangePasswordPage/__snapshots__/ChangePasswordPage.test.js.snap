// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="ChangePasswordPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            Change Password
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={543.5}
                              style={
                                Array [
                                  Object {
                                    "marginTop": 58,
                                  },
                                  undefined,
                                ]
                              }
                              width={543.5}
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "justifyContent": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 43.5,
                                      "lineHeight": 60,
                                      "paddingVertical": 29,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Change your password
                              </Text>
                              <Text
                                allowFontScaling={false}
                                numberOfLines={2}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                To help keep your account secure, we'll send you a new password.
                              </Text>
                            </View>
                            <View
                              accessibilityLabel="Change Password Button"
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#3D3397",
                                  "borderRadius": 10,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                              testID="changePasswordButton"
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Request password reset
                              </Text>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            paddingHorizontal={null}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "bottom": 0,
                                  "paddingHorizontal": 58,
                                  "position": "absolute",
                                  "width": "100%",
                                },
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "translateY": AnimatedValue {
                                        " __value": 0,
                                      },
                                    },
                                  ],
                                },
                              ]
                            }
                            testID="ChangePasswordToast"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#FFFFFF",
                                    "borderColor": "#F2F3F4",
                                    "borderRadius": 10,
                                    "borderWidth": 0,
                                    "padding": 29,
                                    "shadowColor": "#362E60",
                                    "shadowOffset": Object {
                                      "height": 7,
                                      "width": 0,
                                    },
                                    "shadowOpacity": 0.1,
                                    "shadowRadius": 10,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "#0AB075",
                                      "borderColor": "#0AB075",
                                      "borderWidth": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                              type="success"
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "flex-start",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="toastHeader"
                              >
                                <SvgMock
                                  height={36}
                                  width={36}
                                />
                                <Text
                                  allowFontScaling={false}
                                  numberOfLines={2}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 21.5,
                                        "lineHeight": 30.5,
                                        "paddingHorizontal": 18,
                                        "paddingTop": 2,
                                      },
                                      undefined,
                                    ]
                                  }
                                  type="success"
                                >
                                  Password reset e-mail sent
                                </Text>
                                <View
                                  accessibilityLabel="Close toast"
                                  accessibilityRole="button"
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                  testID="toastCloseButton"
                                >
                                  <SvgMock
                                    height={20}
                                    style={
                                      Array [
                                        Object {
                                          "height": 65,
                                          "width": 65,
                                        },
                                        undefined,
                                      ]
                                    }
                                    type="success"
                                    width={20}
                                  />
                                </View>
                              </View>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                      "paddingHorizontal": 54.5,
                                      "paddingTop": 3,
                                    },
                                    undefined,
                                  ]
                                }
                                type="success"
                              >
                                You will receive an e-mail shortly 
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
